version_align.sh

```
@@ -23,8 +23,10 @@ esac
 
 echo "Updating ..."
 
-# Update the version for all crates in the Cargo.toml workspace.dependencies section
-sed "${sedi[@]}" -e '/\[workspace.dependencies\]/,/# Magicblock/s/version = ".*"/version = "='$version'"/' Cargo.toml
+# Update the version pin only on internal `session-keys*` workspace deps. The narrow
+# pattern means third-party deps in the same section (anchor-lang, syn, ...) are never
+# touched, regardless of any fence comment.
+sed "${sedi[@]}" -e '/^session-keys[a-zA-Z0-9_-]* = /s/version = "[^"]*"/version = "='$version'"/' Cargo.toml
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)